modules/detection/lidar_driver.py

```diff
@@ -0,0 +1,270 @@
+        crc = self._create_crc(packet_bytes)
+        packet_bytes.append(crc & 0xFF)
+        packet_bytes.append((crc >> 8) & 0xFF)
+
+        return bytearray(packet_bytes)
+
+    def _parse_packet(self, byte):
+        """
+        Check for packet in byte stream.
+        """
+        if self._packet_parse_state == 0:
+            if byte == 0xAA:
+                self._packet_parse_state = 1
+                self._packet_data = [0xAA]
+
+        elif self._packet_parse_state == 1:
+            self._packet_parse_state = 2
+            self._packet_data.append(byte)
+
+        elif self._packet_parse_state == 2:
+            self._packet_parse_state = 3
+            self._packet_data.append(byte)
+            self._packet_payload_size = (self._packet_data[1] | (self._packet_data[2] << 8)) >> 6
+            self._packet_payload_size += 2
+            self._packet_size = 3
+
+            if self._packet_payload_size > 1019:
+                self._packet_parse_state = 0
+
+        elif self._packet_parse_state == 3:
+            self._packet_data.append(byte)
+            self._packet_size += 1
+            self._packet_payload_size -= 1
+
+            if self._packet_payload_size == 0:
+                self._packet_parse_state = 0
+                crc = self._packet_data[self._packet_size - 2] | (
+                    self._packet_data[self._packet_size - 1] << 8
+                )
+                verify_crc = self._create_crc(self._packet_data[0:-2])
+
+                if crc == verify_crc:
+                    return True
+
+        return False
+
+    def _wait_for_packet(self, port, command, timeout=1):
+        """
+        Wait (up to timeout) for a packet of the specified command to be received.
+        """
+        self._packet_parse_state = 0
+        self._packet_data = []
+        self._packet_payload_size = 0
+        self._packet_size = 0
+
+        end_time = time.time() + timeout
+
+        while True:
+            if time.time() >= end_time:
+                return None
+
+            c = port.read(1)
+
+            if len(c) != 0:
+                b = ord(c)
+                if self.parse_packet(b) is True:
+                    if self._packet_data[3] == command:
+                        return self._packet_data
+
+    def _execute_command(self, port, command, write, data=None, timeout=1):
+        """
+        Send a request packet and wait (up to timeout) for a response.
+        """
+        if data is None:
+            data = []
+
+        packet = self._build_packet(command, write, data)
+        retries = 4
+
+        while retries > 0:
+            retries -= 1
+            port.write(packet)
+
+            response = self._wait_for_packet(port, command, timeout)
+
+            if response is not None:
+                return response
+
+        # pylint: disable=broad-exception-raised
+        raise Exception("LWNX command failed to receive a response.")
+        # pylint: enable=broad-exception-raised
```

Comment:
don't raise an exception, return an indication that the function failed - we should change this function to return True, response, if the response is not none, and False, None, if there is no response or if the date is none